test/allow.js

```diff
@@ -0,0 +1,131 @@
+const assert = require('chai').assert
+const { getUserId, verifyDelegator } = require('../lib/handlers/allow')
+const rdf = require('rdflib')
+const ns = require('solid-namespace')(rdf)
+
+const alice = 'https://alice.com'  // principal
+const agentWebId = 'https://agent.example.com'  // secretary
+
+const verifiedSecretaryGraph = rdf.graph()
+verifiedSecretaryGraph.add(
+    rdf.namedNode(alice),
+    ns.acl('delegates'),
+    rdf.namedNode(agentWebId)
+  )
+const notVerifiedSecretaryGraph = rdf.graph()
+
+describe('allow() handler', () => {
+  describe('getUserId()', () => {
+    const emptyFetchDocument = () => {}
+    it('no webId present', done => {
+      let webId  // no secretary
+      let onBehalfOf = alice  // principal
+      getUserId(webId, onBehalfOf, emptyFetchDocument)
+        .then(userId => {
+          assert.equal(userId, undefined,
+            'Should return undefined when no webId is present')
+          done()
+        })
+    })
+    it('no onBehalfOf present', done => {
+      let webId = alice  // principal
+      let onBehalfOf  // no delegation header
+      getUserId(webId, onBehalfOf, emptyFetchDocument)
+        .then(userId => {
+          assert.equal(webId, userId,
+            'Should return webId when no onBehalfOf is present')
+          done()
+        })
+    })
+    it('throws 500 error if fetchDocument errors', done => {
```

Comment:
I think its an anti-pattern to knowingly allow and test for 500 errors in a web server. If there's a case in which you know you cannot complete a request you've got a slew of semantic error codes to choose from.